BPStatusBarAlert/Classes/BPStatusBarAlert.swift

```diff
@@ -20,6 +21,14 @@ public class BPStatusBarAlert: UIView {
         
     fileprivate var containerWindow: UIWindow?
     
+    fileprivate var deviceStatusBarHeight:CGFloat {
```

Comment:
There are a number of aspects to consider here:

1. Magic numbers or strings (in this case `2436`, `15` and `0`) are generally what we call [an "anti-pattern"][1] and should always be avoided to be embedded in code.
2. An argument can be made that the logic to calculate the height of the status bar is independent of this UIView class that is all about rendering. Conflating responsibilities also makes the code harder to test.

A suggestion to consider is to create another class that encapsulates this logic and [using composition][2] you'd use in this getter function.

```swift
// Create a small class with a "single responsibility" for calculating anything 
// related to a device's dimensions. You can test this class easily in isolation
class DeviceDimensions {
  var iPhoneXDeviceHeight = 2436;
  var notchHeight = 15;

  var statusBarHeight: (CGFloat) {
    get {
      var hasNotch = UIDevice().userInterfaceIdiom == .phone && UIScreen.main.nativeBounds.height == iPhoneXDeviceHeight;
      return hasNotch ? notchHeight : 0;
    }
  }
}

// Using composition, use your new class
public class BPStatusBarAlert: UIView {
  fileprivate var deviceStatusBarHeight:CGFloat {
    var deviceDimensions = new DeviceDimensions();
    return deviceDimensions.statusBarHeight;
  }

  // ...
}
```

[1]: http://sahandsaba.com/nine-anti-patterns-every-programmer-should-be-aware-of-with-examples.html#magic-numbers-and-strings
[2]: https://www.youtube.com/watch?v=wfMtDGfHWpA

Reply:
@hadynz  Thanks for the good comments.

I read the contents and it is a great help to me.

In particular, the DeviceDimensions class makes the constructor easier to test.

@SamehSalama,  Could you push code accepting @hadynz 's comment?

Reply:
@hadynz   Thank you very much for the great article
@ppth0608 for sure, I will accept @hadynz 's comment, however, I would change a little bit, as for DeviceDimensions class, there is no need to add initialization method that accepts parameters of type UIDevice and UIScreen, both are singleton instances, I'd rather let them be computed properties in DeviceDimensions class and make statusBarHeight the only instance property of this class.

Reply:
@SamehSalama Thank you for accepting comment :) And i think updated code is more Awesome than before ^^

However, it seems that some modifications are needed in the current code. (ex. Code Conventions) I will review the code soon. If only this couple is applied, it seems to be possible to merge. Thank you very much for opening PR again.